gateway-js/src/index.ts

```diff
@@ -348,44 +384,79 @@ export class ApolloGateway implements GraphQLService {
         keyHash: options.engine.apiKeyHash,
         graphId: options.engine.graphId,
         graphVariant: options.engine.graphVariant || 'current',
-      }
+      };
     }
 
-    await this.updateComposition();
-    if (
-      (isManagedConfig(this.config) || this.experimental_pollInterval) &&
-      !this.pollingTimer
-    ) {
-      this.pollServices();
-    }
+    this.maybeWarnOnConflictingConfig();
 
-    const mode = isManagedConfig(this.config) ? 'managed' : 'unmanaged';
+    // Handles initial assignment of `this.schema`, `this.queryPlannerPointer`
+    isStaticConfig(this.config)
+      ? this.loadStatic(this.config)
```

Comment:
Is there a particular reason that loadStatic takes config as an argument and loadDynamic reads it from `this`? I mean it's not bad for them to have different interfaces, but if it's actually important that they differ I don't see why.

Reply:
`loadDynamic` no longer reads from config at all, which is why the arg went away.

I prefer `loadStatic` to accept it as an argument since we can refine the type of config to `StaticConfig` on its way in, rather than do another `isStaticConfig(this.config)` once inside the function.

Reply:
Ah, I see — loadDynamic calls functions which read from this.config but doesn't do so directly.

Actually, thinking about this makes me wonder if `Experimental_UpdateServiceDefinitions` should be changed to take a DynamicConfig instead of a GatewayConfig?

Reply:
6ddb960